src/main/java/seedu/mark/model/predicates/FolderContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,32 @@
+package seedu.mark.model.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.mark.commons.util.StringUtil;
+import seedu.mark.model.bookmark.Bookmark;
+
+/**
+ * Tests that part of a {@code Bookmark}'s {@code Folder} matches any of the keywords given.
+ */
+public class FolderContainsKeywordsPredicate implements Predicate<Bookmark> {
+    private final List<String> keywords;
+
+    public FolderContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Bookmark bookmark) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsPhraseIgnoreCase(bookmark.getFolder().folderName, keyword));
```

Comment:
Do you think it'd be better to do an exact match for Folder and Tag names? e.g. if I'm searching for folder `NUS`, I may not want to see bookmarks from folders `NUSMods`, `LumiNUS`, etc.